### PR TITLE
Friendship

### PR DESCRIPTION
**Pull request by martinnajjar12**
### Friendship Feature
In this milestone we:
- Created a Friendship model.
- Associated the new model with the User model.
- Created 'Add Friend', 'Confirm', and 'Decline' friendship buttons.
- Wrote an integration test.

**IMPORTANT NOTE:** Run `yarn install --check-files` in order to run the app properly

### CODE REVIEW

**Review by javitocor — CHANGES_REQUESTED** [review]
# STATUS CHANGES REQUIRED ⭐ 
A great effort in this milestone, team! 🥳 
But we need to implement some changes to get the project approved 
Check my comments below 👇  apply the changes and request for a new code review ⏩ 
Good luck! 😺 
## Mandatory changes
- Your app is not running dua a problem, got this error, kindly fix it
``Sass::SyntaxError at /users/sign_in
File to import not found or unreadable: bootstrap/scss/bootstrap.
Load paths:
  /ror-social-scaffold/app/assets/config
  ror-social-scaffold/app/assets/images
 /ror-social-scaffold/app/assets/javascripts
  ror-social-scaffold/app/assets/stylesheets``
**[OPTIONAL]**
- Improve your readme file,

**Review by teekaytech — CHANGES_REQUESTED** [review]
## Status: Changes Required :warning: :vertical_traffic_light: 

### Hello team :wave:

__You've made laudable progress in this milestone.
However, some changes are required before this project can be approved. Please check my comments below and fix them as appropriate.__

**[ REQUIRED CHANGES ]**

- [x] As mentioned by the previous reviewer, initially I was unable to access your project. I was having this error:

Commenting on this line of code, `@import "bootstrap/scss/bootstrap";` in your `/ror-social-scaffold/app/assets/stylesheets/application.css` file solved the problem. Kindly fix this bug so visitors can seamlessly test and use your app.

- [x] - [ ] Posts timeline includes posts written by **logged-in user and their friends** ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))
- Right now, a logged-in user can see all posts, including their own posts, the ones made by their friends and other users.
- Post timeline should only include posts written by currently logged in user and other users with which they have confirmed friendship.

- [x] Add more unit **and integration** tests! ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/tests.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))




* * *
**Please make the required changes and submit another review request when you are done.**

* * *

**Taofeek Olalere
[Twitter](https://twitter.com/ola_lere) | [Github](https://github.com/teekaytech) | @taofeekolalere on Slack**
__Happy Coding :smiley:__

**Comment by Kingobaino1** [comment]
> * ou

Please to gain access into the app, run the following commands

1. `bundle install`
2. `yarn install --check-files`

**Review by tirthajyoti-ghosh — CHANGES_REQUESTED** [review]
# Changes required 🛑 

## Impressive progress so far 💪! Thanks for making all the changes 👏.

### However, we need to work on some changes before we can merge.

Please add more tests for this milestone. Remember that you are required to add tests for all the models, controllers and features, not just the ones you have created. 

Please add tests for these models and controllers:

- `comment`
- `like`
- `post`
- `user`

Remember to add tests for the model validations and associations. When adding tests for the controllers, you can use request tests but you are free to use feature tests.

Also, don't forget to add both positive and negative scenario tests.

Your project looks absolutely fantastic. Just make these changes and apply for another code review.

Happy coding 👨‍💻 

*Need help? Shoot me a message on Slack - [@Ghosh](https://app.slack.com/client/T47CT8XPG/DRDM6LAQN/user_profile/URB1MHXK6)*
<hr>

**Tirthajyoti Ghosh**

[GitHub](https://github.com/tirthajyoti-ghosh) | [LinkedIn](https://www.linkedin.com/in/tirthajyoti-ghosh/) | [Twitter](https://twitter.com/terrific_ghosh)

File: app/models/user.rb
Comment (ghost) [inline app/models/user.rb:13]: Add tests for the above associations. You can use [shoulda matchers](https://matchers.shoulda.io/) for this.

File: app/models/user.rb
Comment (ghost) [inline app/models/user.rb:44]: Write tests for all the above methods. If however, you have methods that you are not calling outside the `User` class, you can declare them as private and not write tests for them.

**Review by ghost — CHANGES_REQUESTED** [review]
# Status: Changes Required

## Hello team 😄,

**Great job implementing the requested changes 💪. However, you haven't written tests for all your files.**

1) **Please add tests for the following files in `app/models` directory.**

- `post.rb`.
- `like.rb`.
- `comment.rb`.

2). **`spec/helpers` directory is empty. Please add tests for all your helpers in `app/helpers` directory. Hint: If you have any methods that are not being called out of the modules, you can declare them private and not write tests for them.**

3). **Check more in the comments below.**

## [Optional]

- You can user [shoulda matchers](https://matchers.shoulda.io/) to write tests for your validations instead of writing them yourself from scratch.

## Happy coding 🥂 

**PS: You can reach me on slack (Rose Wanjohi).**

**Review by voscarmv — APPROVED** [review]
# Project status: Approved ✨ 😎👍 💯 ✔️
### You may merge this branch and move forward ⛙🛣
***[OPTIONAL]***
- Please add `npm install` in your installation instructions. Otherwise, your page will output a `File to import not found or unreadable: bootstrap/scss/bootstrap.` error on a fresh install.
### Questions or comments? Reach me on [Twitter](https://twitter.com/voscarmv) or [LinkedIn](https://linkedin.com/in/oscar-mier) 🧑‍💻🗨️🧑‍💻 
# Happy Hacking! 🧠 ⌨️ 🖱️ 🖥️

**Pull request merged**
